libs/chain-signatures/contract/src/state/resharing.rs

```diff
@@ -46,9 +46,9 @@ impl ResharingContractState {
         self.resharing_key.epoch_id()
     }
 
-    /// Casts a vote for a re-proposal. Requires the signer to be an original participant from the
-    /// previous running state. If this exceeds the threshold (from the previous running state),
-    /// returns a new ResharingContractState that we should transition into.
+    /// Casts a vote for a re-proposal. Requires the signer to be a participant of the
+    /// prospective epoch.
+    /// Returns a new ResharingContractState if all proposed participants voted for it.
```

Comment:
Just a tip. You can create links in your rust docs, which `rust-analyzer` will also discover and let you go to it's definition from the docs in the editor.

```suggestion
    /// Returns a new [`ResharingContractState`] if all proposed participants voted for it.
```